fdbserver/ClusterController.actor.cpp

```diff
@@ -1489,54 +1489,60 @@ ACTOR Future<Void> statusServer(FutureStream<StatusRequest> requests,
 				}
 			}
 
-			// Get status but trap errors to send back to client.
-			std::vector<WorkerDetails> workers;
-			std::vector<ProcessIssues> workerIssues;
+			while (!requests_batch.empty()) {
+				state std::string requestedStatusJsonField = requests_batch.back().statusField;
```

Comment:
Another idea is to accumulate all the `statusField` in the `requests_batch`, and only issue one `clusterGetStatus` call. After the call returns, find values for each `statusField` and send them back.

Reply:
Yes, I like this suggestion. This kind of resembles what I said in my comment:

> Other approach is to get the entire status json, and then return only the requested fields out of this. But this would make the actor do a lot of unnecessary work in that case.

The benefits of this approach are that:

- we wouldn't need to modify "clusterGetStatus()" (and/or add actors that duplicate code)
- invoke "clusterGetStatus()" once per batch

Disadvantages being:

- we will do the job of populating the entire status (and making the network calls between FoundationDB processes in order to populate the status) even if the caller/batch requests a single field (or a small subset of the entire status)

- we will need to do the additional task of fetching and populating the requested fields from the complete status json output string (but likely this isn't going to add a lot of overhead).

Reply:
Modified code to follow this approach. Please review, thanks!